etstool.py

```diff
@@ -232,8 +239,20 @@ def install(runtime, toolkit, environment, editable, source):
     else:
         commands = ["edm environments create {environment} --force --version={runtime}"]
 
+    # NOTE : Pinning traits to an older version e.g. 6.0 forces older versions
+    # of dependencies e.g. traitsui/pyface < 7.0, which is a problem for
+    # traitsui master because it requires pyface >= 7.0.
+    # So, for now, if an older version of traits is required, we explicitly
+    # allow newer versions of pyface.
+    # This constraint modifier can be removed when we stop supporting traits
+    # 6.0.
+    if TRAITS_VERSION_REQUIRES:
+        constraint_modifiers = CONSTRAINT_MODIFIERS
+    else:
+        constraint_modifiers = ""
```

Comment:
Nit on readability: Could we move this if-else logic together with where `CONSTRAINT_MODIFIERS` is defined? Or just define the constraint modifier here. It is confusing to see `CONSTRAINT_MODIFIERS` being defined as a constant at the top while it is actually a value depending on `TRAITS_VERSION_REQUIRES`.  The `constraint_modifiers` also have different types in these two branches. String happens to be an iterable in Python, but logically, it is a scalar value.

Something like this?
```
CONSTRAINT_MODIFIERS = {"--allow-newer pyface"} if TRAITS_VERSION_REQUIRES else set()
```